network/f5/bigip_gtm_wide_ip.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2015, Michael Perzel
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: bigip_gtm_wide_ip
+short_description: "Manages F5 BIG-IP GTM wide ip"
+description:
+    - "Manages F5 BIG-IP GTM wide ip"
+version_added: "2.0"
+author: 'Michael Perzel'
+notes:
+    - "Requires BIG-IP software version >= 11.4"
+    - "F5 developed module 'bigsuds' required (see http://devcentral.f5.com)"
+    - "Best run as a local_action in your playbook"
+    - "Tested with manager and above account privilege level"
+
+requirements:
+    - bigsuds
+options:
+    server:
+        description:
+            - BIG-IP host
+        required: true
+    user:
+        description:
+            - BIG-IP username
+        required: true
+    password:
+        description:
+            - BIG-IP password
+        required: true
+    lb_method:
+        description:
+            - LB method of wide ip
+        required: true
+        choices: ['return_to_dns', 'null', 'round_robin',
+                      'ratio', 'topology', 'static_persist', 'global_availability',
+                      'vs_capacity', 'least_conn', 'lowest_rtt', 'lowest_hops',
+                      'packet_rate', 'cpu', 'hit_ratio', 'qos', 'bps',
+                      'drop_packet', 'explicit_ip', 'connection_rate', 'vs_score']
+    wide_ip:
+        description:
+            - Wide IP name
+        required: true
+'''
+
+EXAMPLES = '''
+  - name: Set lb method
+    local_action: >
+      bigip_gtm_wide_ip
+      server=192.168.0.1
+      user=admin
+      password=mysecret
+      lb_method=round_robin
+      wide_ip=my_wide_ip
```

Comment:
In version >= 11.6.0 this needs to be a fully qualified domain name without invalid characters (of which underscore is an invalid char). Can we change this to be my-wide-ip.domain.com or something FQDN-y?